### DISCUSSION TITLE
codec priority gets discarded?

### DISCUSSION THREAD

**Discussion by wusspuss** [paullouisageneau/libdatachannel#1142] [Q&A]
I `rtcAddTrack` with an sdp in specific order:
```
import libdatachannel


def onGatheringStateChange(state, pc):
    print("Gathering State:", state)
    if state == libdatachannel.GatheringState.GATHERING_COMPLETE:
        print(pc.local_description)


pc = libdatachannel.PeerConnection()
pc.gathering_state_change_callback = lambda state: onGatheringStateChange(
    state, pc
)
track = pc.add_track(
    "\n".join(
        [
            "audio 9 UDP/TLS/RTP/SAVPF 8 0 9 96",
            "a=rtpmap:8 PCMA/8000",
            "a=rtpmap:0 PCMU/8000",
            "a=rtpmap:9 G722/8000",
            "a=mid:audio",
            "a=sendrecv",
            "a=rtpmap:96 telephone-event/8000",
            "a=rtcp-mux",
        ]
    )
)
pc.set_local_description()
input()
```
and in `rtcGetLocalDescription` that order gets discarded: `m=audio 37834 UDP/TLS/RTP/SAVPF 0 8 9 96`
which makes my peer prefer PCMU, while I was actually asking for PCMA to be preferred. 
Is this by design or am I doing something wrong? Is it possible to disable this sorting?

**Comment by wusspuss** [comment]
Maybe something like [this](https://github.com/paullouisageneau/libdatachannel/pull/1144#issue-2212257333) ?

**Reply by paullouisageneau** [reply]
Indeed, currently priority order gets discarded, thank you for the fix.

**Comment by paullouisageneau** [comment]
Fixed by https://github.com/paullouisageneau/libdatachannel/pull/1168
